src/env.ts

```diff
@@ -250,12 +255,44 @@ export function setEnvConfiguration(config: Configuration, handlers: FunctionInf
     if (config.enableDDTracing !== undefined && environment[ddTracingEnabledEnvVar] === undefined) {
       environment[ddTracingEnabledEnvVar] = config.enableDDTracing;
     }
+
+    if (
+      (config.enableAppSec && config.enableAppSecRuntimeApiProxy) ||
+      (config.enableAppSec && config.enableASM) ||
+      (config.enableASM && config.enableAppSecRuntimeApiProxy)
+    ) {
+      throw new Error(
+        "`enableAppSec`, `enableAppSecRuntimeApiProxy` and `enableASM` are mutually exclusive; set only `enableAppSec` or `enableAppSecRuntimeApiProxy`",
+      );
+    }
+
+    if (config.enableAppSec !== undefined && config.enableAppSec) {
+      if (!config.enableDDTracing) {
+        throw new Error("`enableAppSec` requires `enableDDTracing` to be enabled");
+      }
+      if (type === RuntimeType.PYTHON) {
+        environment[ddAppsecEnabledEnvVar] ??= config.enableAppSec;
+      } else {
+        environment[AWS_LAMBDA_EXEC_WRAPPER_VAR] ??= AWS_LAMBDA_EXEC_WRAPPER;
+        environment[ddServerlessAppsecEnabled] ??= config.enableAppSec;
+      }
+    }
+    if (config.enableAppSecRuntimeApiProxy !== undefined && config.enableAppSecRuntimeApiProxy) {
```

Comment:
Would this work too?
```suggestion
    if (config.enableAppSecRuntimeApiProxy) {
```

Reply:
Absolutely.